demo/test/verifier.cpp

```diff
@@ -30,19 +34,19 @@ class VerifierTest : public ::testing::Test {
         std::make_shared<DummyBehavior>(true, true, goodCommand, "LowPriority");
     MockEnvironmentModel::Ptr environmentModel;
     Time time{Clock::now()};
-    Verifier verifier;
+    Verifier::Ptr verifier;
 };
 
 TEST_F(VerifierTest, basicVerification) {
-    VerificationResult goodResult = verifier.analyze(time, goodCommand);
-    EXPECT_TRUE(goodResult.isOk());
+    AbstractResult::Ptr goodResult = verifier->analyze(time, goodCommand);
+    EXPECT_TRUE(goodResult->isOk());
```

Comment:
Hm… accessing an object named `abstract[…]` is counter-intuitive (you shouldn't be able to instantiate and access an abstact class/object). Maybe we should change the names of our abstract types :see_no_evil: 

```suggestion
    BaseResult::Ptr goodResult = verifier->analyze(time, goodCommand);
    EXPECT_TRUE(goodResult->isOk());
```

Would read more natural and explain what we're doing:
Returning the verification result as a pointer to its base type (which is abstract, yes).

I know this is a bit late, but what do you think?